### PR TITLE
Feature/swagger independence

### PR DESCRIPTION
**Pull request by Not22Alan**
Swagger:
  - seperate folder.
  - switching to dynamice server url.
  - adding components & security schemes.
  - moving all swagger samples to swaggerDocs.

Permissions update:
  - unique id.
  - adding VIEW_ALL_USERS.
  - implementing in routes

Seeders:
  - adding a way to delete.
  - adding roles and permission sedder.
  - updating old seeders.
  - updating package.json.
  - updating readme.md.

### CODE REVIEW

File: routes/api/users.js
Comment (shivam8811) [inline routes/api/users.js:1]: Looks like you have a typing mistake in this file. The middleware name is supposed to be 'checkPermission' and not 'checkPermissions'

File: routes/api/events.js
Comment (shivam8811) [inline routes/api/events.js:1]: Currently the 'checkPermission' middleware checks only for one permission passed as parameter. If you want to check multiple permissions, the middleware should be updated to accept array of permissions and loop through each of them.

File: routes/api/users.js
Comment (Not22Alan) [reply routes/api/users.js:1]: The variable is called checkPermissions and it imports from permissionMiddleware.
It does work, but i get what you mean it's better to have a unified name for all files.

File: routes/api/events.js
Comment (Not22Alan) [reply routes/api/events.js:1]: Done.

**Review by shivam8811 — APPROVED** [review]

**Pull request merged**